### ISSUE TITLE
postgresql support dropped in 4.1.2?

### ISSUE THREAD

**Issue by david-barbion** [haiwen/seafile#1177]
Got this after upgrading to 4.1.2:
```
seafile@xxx:~/seafile-server-latest/seafile/bin$ ./ccnet-server -c /home/seafile/ccnet/ -t
SQLException: Failed to start connection pool -- database protocol 'postgresql' not supported
 raised in ConnectionPool_start at src/db/ConnectionPool.c:287
```

Is it possible to restore postgresql support?

**Comment by shoeper** [comment]
Official postgresql has never been there. There was one user providing necessary changes to successfully run seafile with postgresql.
Think it was @caniwi. Maybe he's also made a script for 4.1.x.

**Comment by david-barbion** [comment]
Hi,

Thanks for you reply.
It is not only for the scripts (installation or upgrade) but for the binaries (like ccnet-server).
I remember of issue #998 (https://github.com/haiwen/seafile/issues/998) with the same issue.

**Comment by shoeper** [comment]
Yeah as I've said. It was never officially supported. So I don't think one of the team tests if postgresql works or not.

**Comment by freeplant** [comment]
This is a build issue. See https://forum.seafile-server.org/t/seafile-server-4-1-2-and-postgresql/1069/8

**Comment by david-barbion** [comment]
After some research, I finally understand which library miss postgresql support. It is located in libzdb.
If the packaging server does not have libpq postgresql shared library, libzdb drops postgresql support.
I installed libpq for my linux system with the command `apt-get install libpq-dev` then libzdb will now add support for postgresql:
```
$ ./configure
[...]
|------------------------------------------------------------|
|   SQLite3:                                      ENABLED    |
|   MySQL:                                        ENABLED    |
|   PostgreSQL:                                   ENABLED    |
|   Oracle:                                       DISABLED   |
+------------------------------------------------------------+
```

My linux system provides libpq.so.5 (postgresql 8.4 and newer) but older version of Seafile are shipped with libpq.so.4. Also, I could not put my new libzdb library with postgresql support in place of seafile's one because of library version mismatch...

In conclusion, for your build server to properly creates libzdb, you should provide the postgresql library to it:
```
wget https://ftp.postgresql.org/pub/source/v9.4.1/postgresql-9.4.1.tar.bz2
tar -xjvf postgresql-9.4.1.tar.bz2 && cd postgresql-9.4.1
./configure --without-readline
make -C src/interfaces install # this will build and install libpq only, dest dir is /usr/local/pgsql/lib
```
Hope this could help to add postgresql support.

**Comment by caniwi** [comment]
Yes, it is the binary support in the build for postgres that is missing.  If we get it re-instated then I can work on the postgresql support.  I have the sql statements ready for conversion of a 4.0.6 instance to 4.1.2.  Please let us know when

**Comment by killing** [comment]
The support will be added back in 4.2, the next release.

2015-05-05 13:52 GMT+08:00 caniwi <notifications@github.com>:
